Plug-Ins/Dialog Editor/DialogEditorWindowController.swift

```diff
@@ -0,0 +1,1156 @@
+			let fillColor = NSColor.white
+			let strokeColor = NSColor.black
+			fillColor.setFill()
+			strokeColor.setStroke()
+			let lozenge = NSBezierPath(roundedRect: self.bounds, xRadius: 8.0, yRadius: 8.0)
+			lozenge.fill()
+			lozenge.stroke()
+			
+			let ps = NSMutableParagraphStyle()
+			ps.alignment = .center
+			let attrs: [NSAttributedString.Key:Any] = [.foregroundColor: strokeColor, .paragraphStyle: ps, .font: NSFontManager.shared.font(withFamily: "Silom", traits: [], weight: 0, size: 12.0)!]
+			let measuredSize = title.size(withAttributes: attrs)
+			var textBox = self.bounds
+			textBox.origin.y -= (textBox.size.height - measuredSize.height) / 2
+			title.draw(in: textBox, withAttributes: attrs)
+		case .checkBox:
+			let fillColor = NSColor.white
+			let strokeColor = NSColor.black
+			fillColor.setFill()
+			strokeColor.setStroke()
+			var box = self.bounds
+			box.size.width = box.size.height
+			box = box.insetBy(dx: 2, dy: 2)
+			NSBezierPath.stroke(box)
+			NSBezierPath.fill(box)
+			
+			title.draw(at: NSPoint(x: box.maxX + 4, y: 0), withAttributes: [.foregroundColor: strokeColor, .font: NSFontManager.shared.font(withFamily: "Silom", traits: [], weight: 0, size: 12.0)!])
+		case .radioButton:
+			let fillColor = NSColor.white
+			let strokeColor = NSColor.black
+			fillColor.setFill()
+			strokeColor.setStroke()
+			var box = self.bounds
+			box.size.width = box.size.height
+			box = box.insetBy(dx: 2, dy: 2)
+			let lozenge = NSBezierPath(ovalIn: box)
+			lozenge.fill()
+			lozenge.stroke()
+			
+			title.draw(at: NSPoint(x: box.maxX + 4, y: 0), withAttributes: [.foregroundColor: strokeColor, .font: NSFontManager.shared.font(withFamily: "Silom", traits: [], weight: 0, size: 12.0)!])
+		case .control:
+			let fillColor = NSColor.white
+			let strokeColor = NSColor.black
+			fillColor.setFill()
+			strokeColor.setStroke()
+			NSBezierPath.fill(self.bounds)
+			NSBezierPath.stroke(self.bounds)
+		case .staticText:
+			title.draw(at: NSZeroPoint, withAttributes: [.foregroundColor: NSColor.black, .font: NSFontManager.shared.font(withFamily: "Silom", traits: [], weight: 0, size: 12.0)!])
+		case .editText:
+			let fillColor = NSColor.white
+			let strokeColor = NSColor.black
+			fillColor.setFill()
+			strokeColor.setStroke()
+			NSBezierPath.fill(self.bounds)
+			NSBezierPath.stroke(self.bounds)
+			
+			title.draw(at: NSZeroPoint, withAttributes: [.foregroundColor: strokeColor, .font: NSFontManager.shared.font(withFamily: "Silom", traits: [], weight: 0, size: 12.0)!])
+		case .icon:
+			var resource: Resource?
+			resource = manager.findResource(type: ResourceType("cicn"), id: resourceID, currentDocumentOnly: true)
+			if resource == nil {
+				resource = manager.findResource(type: ResourceType("ICON"), id: resourceID, currentDocumentOnly: true)
+			}
+			if let resource = resource {
+				var format: UInt32 = 0
+				let imgRep = DITLItemView.imageRep(for: resource, format: &format)
+				imgRep?.draw(in: self.bounds)
```

Comment:
You should be able to use `resource.preview` here rather than replicating code from the image plugin.
```suggestion
				resource.preview {
					$0?.draw(in: self.bounds)
				}			
```